ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot

```diff
@@ -40,6 +40,37 @@ Verify Ods Users Can Do Http Request That Must Be Redirected to Https
     Should Start With    ${url}    https
     Remove Pipeline Project    project-redirect-http
 
+Verify DSPO Operator Reconciliation Retry
+    [Documentation]    Verify DSPO Operator is able to recover from missing components during the initialization
+    [Tags]      Sanity    Tier1
+    ${local_project_name} =    Set Variable    recon-test
+    ${stopped} =    Set Variable    ${False}
+    # limit is 180 because the reconciliation run every 2 minutes
+    ${timeout} =    Set Variable    180
+    New Project    ${local_project_name}
+    Install DataSciencePipelinesApplication CR    ${local_project_name}    data-science-pipelines-reconciliation.yaml    False
+    # wait the CR be ready
+    Wait Until Keyword Succeeds    15 times    1s
+    ...    Run    oc get datasciencepipelinesapplications -n ${local_project_name}
```

Comment:
I think you must use `Run And Return Rc And Output` and after that check  the return code is 0. Otherwise this line won't never fail.

Reply:
`Run` and `Run And Return Rc And Output` are both passing when using an invalid resource. I need more time to investigate. If you know any solution, please let me know

Reply:
Done